pkgs/by-name/kc/kcc/package.nix

```
@@ -0,0 +1,63 @@
+{
+  stdenv,
+  lib,
+  qt6,
+  python3Packages,
+  fetchFromGitHub,
+  p7zip,
+  archiveSupport ? true,
+  nix-update-script,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "kcc";
+  version = "7.0.0";
+
+  src = fetchFromGitHub {
+    owner = "ciromattia";
+    repo = "kcc";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-aFgw34B+bg0fralh97Ae2emOzBz/m/bfVC1VhnCLIE4=";
+  };
+
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+
+  buildInputs = lib.optionals stdenv.hostPlatform.isLinux [ qt6.qtwayland ];
+  propagatedBuildInputs = (
+    with python3Packages;
+    [
+      packaging
+      pillow
+      psutil
+      python-slugify
+      raven
+      requests
+      natsort
+      mozjpeg_lossless_optimization
+      distro
+      pyside6
+      numpy
+    ]
+  );
+
+  qtWrapperArgs = lib.optionals archiveSupport [ ''--prefix PATH : ${lib.makeBinPath [ p7zip ]}'' ];
+
+  postFixup = ''
+    wrapProgram $out/bin/kcc-c2e --prefix PATH : ${lib.makeBinPath [ p7zip ]}
+  '';
+
```

As I was testing things out, I realized that it is possible to do a version check. I found out that `versionCheckHook` is preferred for this over the `passthru`. The hook is documented [here](https://github.com/NixOS/nixpkgs/blob/master/doc/hooks/versionCheckHook.section.md).

```suggestion
  nativeInstallCheckInputs = [ versionCheckHook ];
  versionCheckProgram = "${placeholder "out"}/bin/kcc-c2e";
```

This will verify the version number found in the `--help` flag.